cmd/descheduler/app/server.go

```diff
@@ -94,6 +96,15 @@ func NewDeschedulerCommand(out io.Writer) *cobra.Command {
 				return
 			}
 
+			tracerProvider, err := tracing.NewTracerProvider(s.Tracing.OtelCollector, s.Tracing.OtelTransportCert, s.Tracing.OtelTraceName, s.Tracing.OtelTraceNamespace)
+			if err != nil {
+				klog.Fatalf("failed to setup tracing provider", "err", err)
+				return
+			}
+			traceCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer tracing.Shutdown(traceCtx, tracerProvider)
```

Comment:
we should be re-using the `ctx` that's already here where available rather than creating a new one (noticed this throughout the PR)